alvr/server_core/src/connection.rs

```diff
@@ -265,16 +265,69 @@ pub fn handshake_loop(ctx: Arc<ConnectionContext>, lifecycle_state: Arc<RwLock<L
             manual_client_ips
         };
 
-        if !available_manual_client_ips.is_empty()
-            && try_connect(
+        if !available_manual_client_ips.is_empty() {
+            if available_manual_client_ips.keys().any(|i| i.is_loopback()) {
+                let adb_path = match alvr_adb::get_adb_path() {
+                    Some(path) => path,
+                    None => {
+                        dbg_connection!("Couldn't find adb, installing it...");
+                        if let Err(e) = alvr_adb::install_adb(|downloaded, total| {
+                            dbg_connection!("Downloaded {downloaded:?} bytes of {total:?}");
```

Comment:
This is fine, but maybe make this a `warn!()`, so this will be visible on release mode/default logging settings. Also you should make the message as `"Downloading ADB. {downloaded:?} bytes of {total:?}"`. For the total, display it more nicely, for example put "?" if total is None.